pulp_rpm/app/advisory.py

```diff
@@ -115,8 +115,9 @@ def resolve_advisories(version, previous_version):
         tmp_advisories = []
         for adv in conflict_advisories[1:]:
             pk_to_add, pk_to_remove, pk_to_exclude = resolve_advisory_conflict(new_advisory, adv)
-            new_advisory = UpdateRecord.objects.filter(pk=pk_to_add[0]).get()
-            tmp_advisories.append(new_advisory)
+            if pk_to_add:
+                new_advisory = UpdateRecord.objects.filter(pk=pk_to_add[0]).get()
+                tmp_advisories.append(new_advisory)
```

Comment:
@ggainey This is a new addition right?

Reply:
Yes, this is a final backstop check. It's not new-packages, it's resolve_advisory_conflict() returning no-new-advisory (that pk_ isn't "package, it's "primary key"). I *believe* this was due to the specific data-weirdness on the customer machine, where the rest of the fix here resulted in the decision "between new and incoming we're not actually going to add anything".

Figuring out *exactly* what got us into this state is difficult - the one machine we see this on, we are not allowed to know anything about, due to the very wide/deep security moat it lives on the other side of, it's a black box that we only get stacktraces from.